app/controllers/welcome_controller.rb

```diff
@@ -1,5 +1,91 @@
+                        stats_saving_throws_throws_con: base_info.stats.saving_throws.throws[:con],
+                        stats_saving_throws_throws_int: base_info.stats.saving_throws.throws[:int],
+                        stats_saving_throws_throws_wis: base_info.stats.saving_throws.throws[:wis],
+                        stats_saving_throws_throws_cha: base_info.stats.saving_throws.throws[:cha],
+                        stats_skills_proficiencies: base_info.stats.skills.proficiencies.to_s,
+                        stats_skills_acrobatics: base_info.stats.skills.skills[:acrobatics],
+                        stats_skills_animal_handling: base_info.stats.skills.skills[:animal_handling],
+                        stats_skills_arcana: base_info.stats.skills.skills[:arcana],
+                        stats_skills_athletics: base_info.stats.skills.skills[:athletics],
+                        stats_skills_deception: base_info.stats.skills.skills[:deception],
+                        stats_skills_history: base_info.stats.skills.skills[:history],
+                        stats_skills_insight: base_info.stats.skills.skills[:insight],
+                        stats_skills_intimidation: base_info.stats.skills.skills[:intimidation],
+                        stats_skills_investigation: base_info.stats.skills.skills[:investigation],
+                        stats_skills_medicine: base_info.stats.skills.skills[:medicine],
+                        stats_skills_nature: base_info.stats.skills.skills[:nature],
+                        stats_skills_perception: base_info.stats.skills.skills[:perception],
+                        stats_skills_performance: base_info.stats.skills.skills[:performance],
+                        stats_skills_persuasion: base_info.stats.skills.skills[:persuasion],
+                        stats_skills_religion: base_info.stats.skills.skills[:religion],
+                        stats_skills_sleight_of_hand: base_info.stats.skills.skills[:sleight_of_hand],
+                        stats_skills_stealth: base_info.stats.skills.skills[:stealth],
+                        stats_skills_survival: base_info.stats.skills.skills[:survival],
+                        sub_ancestry: base_info.sub_ancestry.class == String ? base_info.sub_ancestry : base_info.sub_ancestry[:name],
+                        traits: base_info.traits.to_s,
+                        vision: base_info.vision,
+                        is_saved: 0)
+        NpcModelPack.create(npc_model_id: @npc.id, pack_id: base_info.equipment.pack.id)
+        base_info.equipment.armor.each do |piece|
+          NpcModelArmor.create(npc_model_id: @npc.id, armor_id: piece.id) unless piece == ""
+        end
+        base_info.equipment.weapons.each do |weapon|
+          NpcModelWeapon.create(npc_model_id: @npc.id, weapon_id: weapon.id)
+        end
+        @armor = base_info.equipment.armor
+        @pack = [base_info.equipment.pack]
+        @weapons = base_info.equipment.weapons
```

Comment:
This is the BIG change.

I wanted visitors to be able to generate NPCs, but not have those NPCs fill up the DB.  So on the Welcome Index, there is logic to determine whether or not the user is a logged in User or not.  If they are not, as you can see on welcome's `index.html.erb` file, then the @npc is shown in a different way than if they are logged in.  

NPC's that are generated by logged in Users but NOT saved are destroyed when the user logs out.  In a refactor, maybe I'll add that logic in here so that on refresh, the NPC that isn't saved is just destroyed immediately.